### PR TITLE
fix: [M-02] reset `_renounceOwnershipStartedAt` on `acceptOwnership(..)` in LSP14

### PR DESCRIPTION
**Pull request by YamenMerhi**
# What does this PR introduce?
## 🐛 Bug

> From Milo's report

Ownership of LSP0 accounts are transferred through a two-step process.

First, the owner calls [transferOwnership()](https://github.com/lukso-network/lsp-smart-contracts/blob/0c1738619818bcf2e01636f31782886b26fb91d7/contracts/LSP14Ownable2Step/LSP14Ownable2Step.sol#L84-L107) to nominate a pendingOwner:

[LSP14Ownable2Step.sol#L165-L166](https://github.com/lukso-network/lsp-smart-contracts/blob/0c1738619818bcf2e01636f31782886b26fb91d7/contracts/LSP14Ownable2Step/LSP14Ownable2Step.sol#L165-L166)

        _pendingOwner = newOwner;
        delete _renounceOwnershipStartedAt;
Afterwards, the pending owner calls [acceptOwnership()](https://github.com/lukso-network/lsp-smart-contracts/blob/0c1738619818bcf2e01636f31782886b26fb91d7/contracts/LSP14Ownable2Step/LSP14Ownable2Step.sol#L109-L128) to becomes the new owner:

[LSP14Ownable2Step.sol#L176-L177](https://github.com/lukso-network/lsp-smart-contracts/blob/0c1738619818bcf2e01636f31782886b26fb91d7/contracts/LSP14Ownable2Step/LSP14Ownable2Step.sol#L176-L177)

        _setOwner(msg.sender);
        delete _pendingOwner;
However, as _renounceOwnershipStartedAt is only deleted when transferOwnership() is called, LSP0 accounts can be transferred to new owners with a non-zero _renounceOwnershipStartedAt by doing the following:

Call transferOwnership() to nominate a pending owner.
Using execute(), perform a delegate call that overwrites _renounceOwnershipStartedAt to any value.
The pending owner calls acceptOwnership() to gain ownership of the LSP0 account.
This could be problematic depending on what _renounceOwnershipStartedAt is set to.

If it is set to an extremely large value, such as close to type(uint256).max, renounceOwnership() will always revert due to [this check](https://github.com/lukso-network/lsp-smart-contracts/blob/0c1738619818bcf2e01636f31782886b26fb91d7/contracts/LSP14Ownable2Step/LSP14Ownable2Step.sol#L202-L207), making the function DOSed.

If it is set to block.number - RENOUNCE_OWNERSHIP_CONFIRMATION_DELAY, renounceOwnership() will be in the confirmation period for the next 200 blocks. If the new owner accidentally calls renounceOwnership(), he will instantly lose ownership of the account rather than initiate the process.

The fix just ensures a clean state transfer between 2 addresses.

### PR Checklist

- [ ] Wrote Tests
- [ ] Wrote & Generated Documentation (readme/natspec/dodoc)
- [ ] Ran `npm run lint` && `npm run lint:solidity` (solhint)
- [ ] Ran `npm run format` (prettier)
- [ ] Ran `npm run build`
- [ ] Ran `npm run test`

### CODE REVIEW

File: tests/foundry/LSP14Ownable2Step/AcceptOwnershipCleanState.sol [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+contract LSP0Implementation is LSP0ERC725Account {
+    constructor(address _addr) LSP0ERC725Account(_addr) {}
+
+    function renounceOwnershipStartedAt() public view returns (uint256) {
+        return _renounceOwnershipStartedAt;
+    }
+}
+
+contract Implementation {
```
Comment (CJ42) [inline tests/foundry/LSP14Ownable2Step/AcceptOwnershipCleanState.sol:20]: Suggest renaming this contract to something like:

```suggestion
contract LSP14StorageUpdater {
```

To make it more clear below what this contract will be used for:

image

File: tests/foundry/LSP14Ownable2Step/AcceptOwnershipCleanState.sol [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+        // Overwrite _renounceOwnershipAt using a delegatecall
+        Implementation implementation = new Implementation();
+        account.execute(
+            OPERATION_4_DELEGATECALL,
+            address(implementation),
+            0,
+            abi.encodeWithSelector(
+                Implementation.setRenounceOwnershipStartedAt.selector,
+                10000000
```
Comment (CJ42) [inline tests/foundry/LSP14Ownable2Step/AcceptOwnershipCleanState.sol:60]: Suggest separating with `_` for readability

```suggestion
                10_000_000
```

File: tests/foundry/LSP14Ownable2Step/AcceptOwnershipCleanState.sol [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+            )
+        );
+
+        // _renounceOwnershipAt is now set to this value
+        assertEq(account.renounceOwnershipStartedAt(), 10000000);
```
Comment (CJ42) [inline tests/foundry/LSP14Ownable2Step/AcceptOwnershipCleanState.sol:65]: Maybe you could store the `10000000` in a local variable?

```solidity
uint256 newRenounceOwnershipStartedAt = 10_000_000; // number of blocks
```

File: tests/foundry/LSP14Ownable2Step/AcceptOwnershipCleanState.sol [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+        // Will call LSP0's accceptOwnership function that
```
Comment (CJ42) [inline tests/foundry/LSP14Ownable2Step/AcceptOwnershipCleanState.sol:67]: Not sure about the word `Will` what is meant?

```suggestion
        // Calling LSP0's `acceptOwnership()` function must 
        // reset the `_renounceOwnershipStartedAt` state variable
```

**Review by CJ42 — APPROVED** [review]
Looks good 👌 
I have added few suggested changes for the tests

**Review by skimaharvey — APPROVED** [review]

**Review by b00ste — APPROVED** [review]

**Pull request merged**